### PR TITLE
Bug 1812251 - Make the URL icons from the Bookmarks page large

### PR DESCRIPTION
**Pull request by gitstart**
### What
The PR makes the URL icons from the Bookmarks page larger

### Steps to test

1. Have Google Accessibility Scanner installed.
2. Save some bookmarks.
3. Open "Bookmarks" from the three-dot menu.
4. Using the Accessibility Scanner, scan the "Bookmarks" page.

### Screenshots
| Issue | Fix |
| ----- | --- |
| URLicon | 66 Fix |

### Pull Request checklist
 - [ ] Quality: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
 - [ ] Tests: This PR includes thorough tests or an explanation of why it does not
 - [ ] Changelog: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
 - [ ] Accessibility: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features
### After merge
- Milestone: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- Breaking Changes: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1812251

### CODE REVIEW

File: fenix/app/src/main/res/values/dimens.xml [hunk 1]
```diff
@@ -13,7 +13,7 @@
     <dimen name="mozac_browser_menu2_corner_radius">8dp</dimen>
     <dimen name="library_item_height">56dp</dimen>
     <dimen name="library_item_icon_margin_horizontal">16dp</dimen>
-    <dimen name="history_favicon_width_height">40dp</dimen>
+    <dimen name="history_favicon_width_height">48dp</dimen>
```
Comment (Mugurell) [inline fenix/app/src/main/res/values/dimens.xml:16]: If the problem relates to accessibility not to how the UI looks then we should avoid changing the UI and only update the accessibility support.
The touch target can be made bigger by using the [increaseTapArea](https://searchfox.org/mozilla-mobile/rev/2d9dd7b862d87ac539a52b9b0d99160b61745dae/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/ext/View.kt#22-31) method. Example: https://github.com/mozilla-mobile/firefox-android/blob/a05ece3df1f60e7b10ab4b564b82f47b7ba8d0eb/fenix/app/src/main/java/org/mozilla/fenix/library/LibrarySiteItemView.kt#L47

**Review by Mugurell — CHANGES_REQUESTED** [review]

**Comment by gitstart** [comment]
Thank you @Mugurell for reviewing🙏.
Will soon push the requested changes. 🚀

**Comment by gitstart** [comment]
The remaining accessibility issues includes a `Text scaling` issue on the toolbar which will require changing the toolbar UI for the whole app. The second issue is an `Item description` issue on the menu overflow button, because the screen can have multiple overflow buttons which have the same `content description` which will not affect the user as they are on different items.

**Comment by gitstart** [comment]
Hello @Mugurell,
Requested Changes Pushed. 🚀
Thank you 🙏

File: fenix/app/src/main/java/org/mozilla/fenix/library/LibrarySiteItemView.kt [hunk 2]
```diff
@@ -41,6 +41,11 @@ class LibrarySiteItemView @JvmOverloads constructor(
 
     val overflowView: ImageButton get() = binding.overflowMenu
 
+    init {
+        overflowView.increaseTapArea(OVERFLOW_EXTRA_DIPS)
```
Comment (Mugurell) [inline fenix/app/src/main/java/org/mozilla/fenix/library/LibrarySiteItemView.kt:45]: The accessibility issues with this button was resolved on all library screens by https://github.com/mozilla-mobile/firefox-android/pull/1122 which added the padding needed to have the button bigger but visually appear the same.
So this line won't be needed.

File: fenix/app/src/main/java/org/mozilla/fenix/library/LibrarySiteItemView.kt [hunk 3]
```diff
@@ -99,4 +104,8 @@ class LibrarySiteItemView @JvmOverloads constructor(
     enum class ItemType {
         SITE, FOLDER
     }
+
+    companion object {
+        private const val OVERFLOW_EXTRA_DIPS = 16
```
Comment (Mugurell) [inline fenix/app/src/main/java/org/mozilla/fenix/library/LibrarySiteItemView.kt:109]: I see this View already having 40dp height/width - [the favicon](https://searchfox.org/mozilla-mobile/rev/2539bc45c927cd8ec585928216685141b6e7ec25/firefox-android/fenix/app/src/main/res/layout/library_site_item.xml#14-29) uses [history_favicon_width_height](https://searchfox.org/mozilla-mobile/rev/2539bc45c927cd8ec585928216685141b6e7ec25/firefox-android/fenix/app/src/main/res/values/dimens.xml#16).
So we'd only need to increase the tap area with (48 - 40) / 2 => 4dp.
(Want to also note again that we need this only for the favicon, not for the overflow icon)

**Review by Mugurell — CHANGES_REQUESTED** [review]
Thank you for the updates!
There two small things to take care of and then we can look into landing this.

**Comment by gitstart** [comment]
Hello @Mugurell,
Requested changes are pushed. 🙏

**Comment by titooan** [comment]
Thanks for your contribution! It looks like this bug has already been fixed by PR #862, so I'll go ahead and close this. 
Happy to accept more patches in the future!

**Comment by titooan** [comment]
Thanks for your contribution! It looks like this bug has already been fixed by PR #862, so I'll go ahead and close this.
Happy to accept more patches in the future!

**Pull request closed without merging**